### DISCUSSION TITLE
Loose implementation of fast markers to load upwards of 30k markers

### DISCUSSION THREAD

**Discussion by jesussmile** [fleaflet/flutter_map#1481] [Show and tell]
Hi Guys, 

based off of https://github.com/KanarekApp/flutter_map_fast_markers I have written a new `MarkerX`  class that can load huge number of marker points and show them on the map. In my example i am using a QuadTree to better sort the markers and using the bounds of the visible screen to populate the markers. simple logic and the draw on canvas feature that the original author uses (for now) [ will try with prebuilt markers in future] 

Initially I tried to use zoom and populate markers based off of it but It would only do so where my screen was visible when i scrolled up down right left.. nothing.[ there is a markerzoom class if you want to try it out]

So, I decided to go with bounds instead.
I feel it's doing a good job, surely will lag a bit when it has 30k markers in memory, However I am not sure if its just a placebo or is actually working, upto you guys to tell.

I have tried it with few markers may be 5000, works smooth.

Apologies, my knowledge is limited and this is as far as I could go, hopefully one of you can look into this matter and let us know if this is any good ?

If it works i would like to create a plugin.
Never created a plugin so.. yeah!. Hopefully to learn from one of you on how to do it ?

If its similar to what we have now with the default marker, please disregards. Just a poor attempt I guess..

If not then cool .. just wanted to contribute.. :P

There are two options to check in the drop down menu, 

1.short waypoint around 5000 markers

2.Long waypoint around 30000 markers

this is the repo, feel free to try https://github.com/jesussmile/markerx

https://user-images.githubusercontent.com/11044978/230053817-706ed034-f96b-4e7b-ba57-9b0c79dc8511.mp4

https://user-images.githubusercontent.com/11044978/230025056-f118d3ba-21cd-471b-8324-b65f6d9aa36b.mp4

Note- I dont know anyting about PR or how to incorporate it, so.. if you make one tell me what to do next .. haha.

**Comment by JaffaKetchup** [comment]
Hi @jesussmile,

This certainly looks promising, however we can't really accept it into the core at this time.
The problem with using canvas (we've looked into similar concepts in the past) is that it is much more difficult to use, and doesn't hold up once you start trying to allow normal use of widgets. If you can get it to perform better than in the current example app, and use widgets, we'd definitely be interested in incorporating this.

The best option is probably to make a plugin. It's just like [making a normal package](https://docs.flutter.dev/development/packages-and-plugins/developing-packages), but with a few more steps for us to add it to the plugins catalogue. See https://docs.fleaflet.dev/plugins/making-a-plugin for more details. Once you've got a working package and some good documentation, I'd be happy to add it to the list.

Thanks for your contribution, happy to help more if you need it.

**Comment by jesussmile** [comment]
hi @JaffaKetchup 
Could you please clarify if you can remember what issue it had with 'not being able to use with normal widgets?' . There are some issue yes! one is the markers 
identifier will keep changing, was this an issue ? hopefully we can figure out a fix, I tried to use an identifier with markers and it seems to perform better than i expected 

https://user-images.githubusercontent.com/11044978/230111964-9c8a3026-bae1-44cc-a097-aab54293b8ff.mp4

**Reply by JaffaKetchup** [reply]
Sure. I mean that you must paint onto a canvas instead of just specifying a normal `child`. See:
https://github.com/jesussmile/markerx/blob/9a2c1080b50646cbcae1b6add70ae5f46dbb4cdd/lib/markerX.dart#L10-L33

The user must use the `onDraw` callback instead of just being able to specify a `child`.

**Comment by jesussmile** [comment]
Ah I see, I was also thinking the same .. that Draw method is actually expensive, let me try with an svg asset and come back to you. Thank you for the tip.

**Comment by jesussmile** [comment]
https://user-images.githubusercontent.com/11044978/230180560-7b657bff-e321-47aa-913f-995aa85f48bd.mp4

I guess my work is complete with it.  I got the markers to load without much lag. Definitely better than the default markers in terms of speed. Yes you can use a child widget like svg assets etc. Tried it. Works. But I prefer to draw on canvas. 
The only think remaining was the `onTap` functionality, which is finally working as well.. so I guess I am finally done with this issue. I will try to make a plugin. Not sure if it will be useful to anyone. Time to learn how to make plugins lol! Thank you @JaffaKetchup

**Reply by JaffaKetchup** [reply]
Np, let me know once you've created it!
